### DISCUSSION TITLE
Comment récupérer la metadata d'autres joueurs ?

### DISCUSSION THREAD

**Discussion by Zocaxu** [SOZ-Faut-etre-Sub/SOZ-FiveM-Server#54] [Q&A] [answered]
Bonjour !

En essayant d'apprendre le fonctionnement du soz-core m'est venu une question : comment, depuis le client d'un joueur X, accéder à l'état ou la metadata d'un joueur Y qui est stockée sur le serveur ?
En regardant ce qui a déjà été fait, j'ai vu deux possibilités :
- dans le client du joueur X, appeler `Player(targetSource).state.letatquejeveux` qui est écrit et lu côté client et serveur librement de ce que j'ai compris.
- dans le client du joueur X, trigger un RPC event, qui appelle le serveur et qui renvoie la donnée souhaitée à partir de la `PlayerData` du joueur Y.

Quelle est la différence ? Y a-t-il une solution mieux que l'autre ? Une solution spécifique au soz-core ?
Merci beaucoup d'avance ! :D

**Comment by joelwurtz** [comment, marked as answer]
Normalement tu peux souscrire a des evenements pour avoir les données mis a jour du joueur dans chaque module (voir le code dans certains modules)

Pour le state du joueur on évite généralement, car cette donnée est repliqué entre tous les clients et donc si on y met trop de choses on a potentiellement beaucoup d'appels réseau et pas vraiment bon pour le serveur / petites connections, on utilise ce principe uniquement quand un joueur a besoin d'une info sur un autre joueur coté client.

Et même dans ce cas on évite, on prefère faire un maximums de calculs et de vérification coté serveur pour éviter tout problème de triche car c'est la source de donnée la plus fiable, donc generlament on va essaye de construire un maximum de logique entre joueurs du coté serveur et utilise seulement le client pour recuperer ces informations calculés et les affichés ou déduire des actions dans le jeu.

**Reply by Zocaxu** [reply]
Merci beaucoup pour la réponse. J'avais bien compris qu'on privilégie le serveur pour les données des joueurs. Dans certains cas (par exemple, un médecin a besoin de savoir si on joueur est en ITT par exemple) le client a besoin de lire la metadata des autres joueurs occasionnellement. Dans ce cas j'imagine que c'est préférable d'utiliser des events Rpc qui vont simplement lire l'état du joueur.

**Reply by joelwurtz** [reply]
Dans ce cas la oui le RPC est mieux, car tous les joueurs n'ont pas besoin de l'information
